flagsmith-engine/segments/models.ts

```diff
@@ -60,10 +66,12 @@ export class SegmentConditionModel {
     matchesTraitValue(traitValue: any) {
         const evaluators: { [key: string]: CallableFunction } = {
             evaluateNotContains: (traitValue: any) => {
-                return !traitValue.includes(this.value);
+                return !traitValue.includes(this.value)
             },
             evaluateRegex: (traitValue: any) => {
-                return !!traitValue.match(new RegExp(this.value));
+                if (this.value != null || this.value != undefined){
```

Comment:
Good catch here, without this we could have ended up with some uncaught `TypeError`s. I think, however, that `null == undefined` so you could get away with only including one of these as far as I'm aware. If you want to explicitly check that something isn't `null` (but handle `undefined` differently) you'd need to do something like `this.value !== null` (although I don't think this is the correct solution here).